### PR TITLE
add zh-trans:docs/setup/production-environment/windows/user-guide-win…

### PR DESCRIPTION
**Pull request by IreneByron**
add zh-trans:docs/setup/production-environment/windows/user-guide-windows-containers.md

### CODE REVIEW

File: content/zh/docs/setup/production-environment/windows/user-guide-windows-containers.md [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+* 重要的是要注意，对于 Linux 和 Windows 容器，在 Kubernetes  上创建和部署服务和工作负载的行为几乎相同。与集群接口的  [Kubectl  命令](/docs/reference/kubectl/overview/)相同。提供以下部分中的示例只是为了快速启动  Windows 容器的使用体验。
+
+<!--
+## Getting Started: Deploying a Windows container
+
+To deploy a Windows container on Kubernetes, you must first create an example application. The example YAML file below creates a simple webserver application. Create a service spec named `win-webserver.yaml` with the contents below:
+-->
+## 入门：部署 Windows 容器
+
+要在 Kubernetes 上部署 Windows 容器，您必须首先创建一个示例应用程序。下面的示例 YAML  文件创建了一个简单的 Web 服务器应用程序。创建一个名为  `win-webserver.yaml`  的 service spec，其内容如下：
```
Comment (tengqm) [inline content/zh/docs/setup/production-environment/windows/user-guide-windows-containers.md]:
```suggestion
要在 Kubernetes 上部署 Windows 容器，您必须首先创建一个示例应用程序。下面的示例 YAML  文件创建了一个简单的 Web 服务器应用程序。创建一个名为  `win-webserver.yaml`  的服务规约，其内容如下：
```

File: content/zh/docs/setup/production-environment/windows/user-guide-windows-containers.md [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+     nodeSelector:
+      beta.kubernetes.io/os: windows
+```
+
+{{< note >}}
+<!--
+Port mapping is also supported, but for simplicity in this example the container port 80 is exposed directly to the service.
+-->
+还支持端口映射，但为简单起见，在此示例中，容器端口80直接暴露给服务。
```
Comment (tengqm) [inline content/zh/docs/setup/production-environment/windows/user-guide-windows-containers.md]:
```suggestion
端口映射也是支持的，但为简单起见，在此示例中容器端口 80 直接暴露给服务。
```

File: content/zh/docs/setup/production-environment/windows/user-guide-windows-containers.md [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+    ```
+
+1. 部署服务并观察 pod 更新：
+
+    ```bash
+    kubectl apply -f win-webserver.yaml
+    kubectl get pods -o wide -w
+    ```
+
+    正确部署服务后，两个 Pod 都标记为“就绪”。要退出 watch 命令，请按 Ctrl + C。
```
Comment (tengqm) [inline content/zh/docs/setup/production-environment/windows/user-guide-windows-containers.md]:
```suggestion
    正确部署服务后，两个 Pod 都标记为“Ready”。要退出 watch 命令，请按 Ctrl + C。
```
```suggestion
    正确部署服务后，两个 Pod 都标记为“就绪”。要退出 watch 命令，请按 Ctrl + C。
```

File: content/zh/docs/setup/production-environment/windows/user-guide-windows-containers.md [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+
+1. 检查部署是否成功。验证：
+
+    * Windows 节点上每个 Pod 有两个容器，使用  `docker ps` 
+    * Linux 主机列出两个 Pod，使用  `kubectl get pods` 
+    * 跨网络的节点到 Pod 通信，从 Linux 主服务器 `curl` 您的 pod IPs 的端口80，以检查 Web 服务器响应
+    * Pod 到 Pod 的通信，使用 docker exec 或 kubectl exec 在pod之间（以及跨主机，如果您有多个 Windows 节点）进行 ping 操作
```
Comment (tengqm) [inline content/zh/docs/setup/production-environment/windows/user-guide-windows-containers.md]:
```suggestion
    * Pod 到 Pod 的通信，使用 docker exec 或 kubectl exec 在 Pod 之间（以及跨主机，如果您有多个 Windows 节点）进行 ping 操作
```

File: content/zh/docs/setup/production-environment/windows/user-guide-windows-containers.md [hunk 5]
```diff
@@ -0,0 +1,244 @@
...
+    * 服务到 Pod 的通信，从 Linux 主服务器和各个 Pod 中 `curl` 虚拟服务 IP（在 `kubectl get services` 下可见）
+    * 服务发现，使用 Kubernetes `curl` 服务名称[默认 DNS  后缀](/docs/concepts/services-networking/dns-pod-service/#services)
```
Comment (tengqm) [inline content/zh/docs/setup/production-environment/windows/user-guide-windows-containers.md]:
```suggestion
    * 服务发现，使用 Kubernetes `curl` 服务名称[默认 DNS 后缀](/docs/concepts/services-networking/dns-pod-service/#services)
```

File: content/zh/docs/setup/production-environment/windows/user-guide-windows-containers.md [hunk 6]
```diff
@@ -0,0 +1,244 @@
...
+    * 入站连接，从 Linux 主服务器或集群外部的计算机 `curl` NodePort
+    * 出站连接，使用 kubectl exec 从 pod 内部 curl 外部 IP
```
Comment (tengqm) [inline content/zh/docs/setup/production-environment/windows/user-guide-windows-containers.md]:
```suggestion
    * 出站连接，使用 kubectl exec 从 Pod 内部 curl 外部 IP
```

File: content/zh/docs/setup/production-environment/windows/user-guide-windows-containers.md [hunk 7]
```diff
@@ -0,0 +1,244 @@
...
+{{< /note >}}
+
+<!--
+## Using configurable Container usernames
+
+Starting with Kubernetes v1.16, Windows containers can be configured to run their entrypoints and processes with different usernames than the image defaults. The way this is achieved is a bit different from the way it is done for Linux containers. Learn more about it [here](/docs/tasks/configure-pod-container/configure-runasusername/).
+-->
+## 使用可配置的容器用户名
+
+从 Kubernetes v1.16 开始，可以为 Windows  容器配置与其镜像默认值不同的用户名来运行其入口点和进程。实现此方法的方式和对 Linux 容器的处理方式有些不同。在[此处](/docs/tasks/configure-pod-container/configure-runasusername/)了解更多信息。
```
Comment (tengqm) [inline content/zh/docs/setup/production-environment/windows/user-guide-windows-containers.md]:
```suggestion
从 Kubernetes v1.16 开始，可以为 Windows  容器配置与其镜像默认值不同的用户名来运行其入口点和进程。此能力的实现方式和 Linux 容器有些不同。在[此处](/docs/tasks/configure-pod-container/configure-runasusername/)可了解更多信息。
```

File: content/zh/docs/setup/production-environment/windows/user-guide-windows-containers.md [hunk 8]
```diff
@@ -0,0 +1,244 @@
...
+Starting with Kubernetes v1.14, Windows container workloads can be configured to use Group Managed Service Accounts (GMSA). Group Managed Service Accounts are a specific type of Active Directory account that provides automatic password management, simplified service principal name (SPN) management, and the ability to delegate the management to other administrators across multiple servers. Containers configured with a GMSA can access external Active Directory Domain resources while carrying the identity configured with the GMSA. Learn more about configuring and using GMSA for Windows containers [here](/docs/tasks/configure-pod-container/configure-gmsa/).
+-->
+## 使用组托管服务帐户管理工作负载身份
+
+从 Kubernetes v1.14 开始，可以将 Windows 容器工作负载配置为使用组托管服务帐户（GMSA）。组托管服务帐户是 Active Directory 帐户的一种特定类型，它提供自动密码管理，简化的服务主体名称（SPN）管理以及将管理委派给跨多台服务器的其他管理员的功能。配置了 GMSA 的容器可以访问外部 Active Directory 域资源，同时携带通过 GMSA 配置的身份。在[此处](/docs/tasks/configure-pod-container/configure-gmsa/)了解有关为 Windows 容器配置和使用 GMSA 的更多信息。
+
+<!--
+## Taints and Tolerations
+-->
+## Taints 和 Tolerations
```
Comment (tengqm) [inline content/zh/docs/setup/production-environment/windows/user-guide-windows-containers.md]:
```suggestion
## 污点和容忍度
```

File: content/zh/docs/setup/production-environment/windows/user-guide-windows-containers.md [hunk 9]
```diff
@@ -0,0 +1,244 @@
...
+
+<!--
+Users today need to use some combination of taints and node selectors in order to keep Linux and Windows workloads on their respective OS-specific nodes. This likely imposes a burden only on Windows users. The recommended approach is outlined below, with one of its main goals being that this approach should not break compatibility for existing Linux workloads.
+-->
+如今，用户需要使用 taints 和 node selector 的某种组合，以便将 Linux 和 Windows 工作负载保留在各自的特定操作系统的节点上。这可能仅对 Windows 用户造成负担。推荐的方法概述如下，其主要目标之一是该方法不应破坏与现有 Linux 工作负载的兼容性。
```
Comment (tengqm) [inline content/zh/docs/setup/production-environment/windows/user-guide-windows-containers.md]:
```suggestion
目前，用户需要将 Linux 和 Windows 工作负载运行在各自特定的操作系统的节点上，因而需要结合使用污点和节点选择算符。这可能仅给 Windows 用户造成不便。推荐的方法概述如下，其主要目标之一是该方法不应破坏与现有 Linux 工作负载的兼容性。
```

File: content/zh/docs/setup/production-environment/windows/user-guide-windows-containers.md [hunk 10]
```diff
@@ -0,0 +1,244 @@
...
+
+<!--
+### Ensuring OS-specific workloads land on the appropriate container host
+-->
+### 确保特定操作系统的工作负载落在适当的容器主机上
+
+<!--
+Users can ensure Windows containers can be scheduled on the appropriate host using Taints and Tolerations. All Kubernetes nodes today have the following default labels:
+-->
+用户可以使用 Taints 和 Tolerations 确保 Windows 容器可以调度在适当的主机上。今天，所有 Kubernetes 节点都具有以下默认标签：
```
Comment (tengqm) [inline content/zh/docs/setup/production-environment/windows/user-guide-windows-containers.md]:
```suggestion
用户可以使用污点和容忍度确保 Windows 容器可以调度在适当的主机上。目前所有 Kubernetes 节点都具有以下默认标签：
```

File: content/zh/docs/setup/production-environment/windows/user-guide-windows-containers.md [hunk 11]
```diff
@@ -0,0 +1,244 @@
...
+
+* beta.kubernetes.io/os = [windows|linux]
+* beta.kubernetes.io/arch = [amd64|arm64|...]
+
+<!--
+If a Pod specification does not specify a nodeSelector like `"beta.kubernetes.io/os": windows`, it is possible the Pod can be scheduled on any host, Windows or Linux. This can be problematic since a Windows container can only run on Windows and a Linux container can only run on Linux. The best practice is to use a nodeSelector.
+-->
+如果 Pod 规范未指定诸如 `"beta.kubernetes.io/os": windows` 之类的 nodeSelector，则可能会将 Pod 调度在任何主机（Windows 或 Linux）上。这是有问题的，因为 Windows 容器只能在 Windows 上运行，而 Linux 容器只能在 Linux 上运行。最佳实践是使用 nodeSelector。
```
Comment (tengqm) [inline content/zh/docs/setup/production-environment/windows/user-guide-windows-containers.md]:
```suggestion
如果 Pod 规范未指定诸如 `"beta.kubernetes.io/os": windows` 之类的 nodeSelector，则该 Pod 可能会被调度到任何主机（Windows 或 Linux）上。这是有问题的，因为 Windows 容器只能在 Windows 上运行，而 Linux 容器只能在 Linux 上运行。最佳实践是使用 nodeSelector。
```

File: content/zh/docs/setup/production-environment/windows/user-guide-windows-containers.md [hunk 12]
```diff
@@ -0,0 +1,244 @@
...
+
+<!--
+However, we understand that in many cases users have a pre-existing large number of deployments for Linux containers, as well as an ecosystem of off-the-shelf configurations, such as community Helm charts, and programmatic Pod generation cases, such as with Operators. In those situations, you may be hesitant to make the configuration change to add nodeSelectors. The alternative is to use Taints. Because the kubelet can set Taints during registration, it could easily be modified to automatically add a taint when running on Windows only.
+-->
+但是，我们了解到，在许多情况下，用户都有既存的大量的 Linux 容器部署，以及一个现成的配置生态系统，例如社区 Helm charts，以及程序化 Pod 生成案例，例如运营商。在这些情况下，您可能会不愿意进行配置更改以添加 nodeSelector。替代方法是使用 Taints。由于 kubelet 可以在注册期间设置 Taints，因此可以轻松修改它，使其仅在 Windows 上运行时自动添加 taint。
```
Comment (tengqm) [inline content/zh/docs/setup/production-environment/windows/user-guide-windows-containers.md]:
```suggestion
但是，我们了解到，在许多情况下，用户都有既存的大量的 Linux 容器部署，以及一个现成的配置生态系统，例如社区 Helm charts，以及程序化 Pod 生成案例，例如 Operators。在这些情况下，您可能会不愿意更改配置添加 nodeSelector。替代方法是使用污点。由于 kubelet 可以在注册期间设置污点，因此可以轻松修改它，使其仅在 Windows 上运行时自动添加污点。
```

File: content/zh/docs/setup/production-environment/windows/user-guide-windows-containers.md [hunk 13]
```diff
@@ -0,0 +1,244 @@
...
+
+<!--
+For example:  `--register-with-taints='os=Win1809:NoSchedule'`
+-->
+例如：`--register-with-taints ='os = Win1809：NoSchedule'`
```
Comment (tengqm) [inline content/zh/docs/setup/production-environment/windows/user-guide-windows-containers.md]:
```suggestion
例如：`--register-with-taints='os=Win1809:NoSchedule'`
```

File: content/zh/docs/setup/production-environment/windows/user-guide-windows-containers.md [hunk 14]
```diff
@@ -0,0 +1,244 @@
...
+
+<!--
+By adding a taint to all Windows nodes, nothing will be scheduled on them (that includes existing Linux Pods). In order for a Windows Pod to be scheduled on a Windows node, it would need both the nodeSelector to choose Windows, and the appropriate matching toleration.
+-->
+通过向所有 Windows 节点添加 taint，将不会在它们上调度任何东西（包括现有的 Linux Pod）。为了使 Windows Pod 调度在 Windows 节点上，它既需要 nodeSelector 选择 Windows，也需要适当的匹配 toleration。
```
Comment (tengqm) [inline content/zh/docs/setup/production-environment/windows/user-guide-windows-containers.md]:
```suggestion
向所有 Windows 节点添加污点后，Kubernetes 将不会在它们上调度任何负载（包括现有的 Linux Pod）。为了使某 Windows Pod 调度到 Windows 节点上，该 Pod 既需要 nodeSelector 选择 Windows，也需要合适的匹配的容忍度设置。
```

**Comment by tengqm** [comment]
There are still some items not resolved.

**Comment by IreneByron** [comment]
@tengqm  All comments has solved yet.

**Comment by tengqm** [comment]
Thanks.
/lgtm
/approve

**Pull request merged**